app/assets/src/components/views/compare/SamplesHeatmapView/SamplesHeatmapView.jsx

```diff
@@ -456,6 +460,64 @@ class SamplesHeatmapView extends React.Component {
     };
   }
 
+  fetchBackgroundData() {
+    if (this.lastRequestToken)
+      this.lastRequestToken.cancel("Parameters changed");
+    this.lastRequestToken = axios.CancelToken.source();
+
+    return updateHeatmapBackground(
+      {
+        sampleIds: this.state.sampleIds,
+        taxonIds: Array.from(this.state.allTaxonIds),
+        removedTaxonIds: Array.from(this.removedTaxonIds),
+        background: this.state.selectedOptions.background,
+        heatmapTs: this.props.heatmapTs,
+      },
+      this.lastRequestToken.token
+    );
+  }
+
+  async fetchBackground() {
+    this.setState({ loading: true });
+    let heatmapData = await this.fetchBackgroundData();
+    let newState = this.extractBackgroundMetrics(heatmapData);
+
+    this.updateHistoryState();
+    this.setState(newState, this.updateFilters);
+  }
+
+  extractBackgroundMetrics(rawData) {
+    let { sampleDetails, allTaxonDetails, allData } = this.state;
+
+    const backgroundMetrics = [
```

Comment:
Does this need to be defined every time? Should be a constant no?

Reply:
Moved to a constant!